worker/transcribee_worker/worker.py

```diff
@@ -122,18 +132,28 @@ async def perform_task(self, task: AssignedTask):
         else:
             raise ValueError(f"Invalid task type: '{task.task_type}'")
 
-    async def _init_doc(self, document_id: str, doc: automerge.Document):
-        with automerge.transaction(doc, "Initialize Document") as d:
-            if d.children is None:
-                d.children = []
-            if d.diarization is None:
-                d.diarization = []
-            if d.speaker_names is None:
-                d.speaker_names = {}
+    async def _preprocess_doc(self, document_id: str, doc: automerge.Document):
+        if doc.version is None:
+            if automerge.dump(doc) == {}:
+                with automerge.transaction(doc, "Initialize Document") as d:
+                    if d.children is None:
+                        d.children = []
+                    if d.diarization is None:
+                        d.diarization = []
+                    if d.speaker_names is None:
+                        d.speaker_names = {}
+                    d.version = 1
+
+                change = d.get_change()
+                if change is not None:
+                    await self.send_change(document_id, change.bytes())
+            else:
+                raise UnsupportedDocumentVersion()
+
+        if doc.version != 1:
+            raise UnsupportedDocumentVersion()
 
-        change = d.get_change()
-        if change is not None:
-            await self.send_change(document_id, change.bytes())
+        return doc
 
     async def get_document_state(self, document_id: str) -> automerge.Document:
```

Comment:
Maybe for another time: But this should return the transcribee Document type

Reply:
But it doesn't, it's still an automerge document and we need it to be. I think @rroohhh was working on better typing for the automerge-library that would allow us to give the proper typehints for this

Reply:
So for another time :)